classes/controllers/grid/GridHandler.inc.php

```diff
@@ -390,7 +392,9 @@ public function setGridDataElements($data)
     {
         $this->callFeaturesHook('setGridDataElements', ['grid' => &$this, 'data' => &$data]);
 
-        if (is_iterable($data)) {
+        if ($data instanceof Enumerable) {
+            $this->_data = EntityDAO::toAssociativeArray($data);
```

Comment:
Although Alec initially attached the code to `EntityDAO`, we agreed to move it to `GridHandler`. So you should move the `toAssociativeArray` method off of `EntityDAO` and into `GridHandler`.

Reply:
Moved `toAssociativeArray ` to `GridHandler` and changed the code.